deepchem/feat/graph_data.py

```diff
@@ -333,12 +334,34 @@ def __init__(self, graph_list: Sequence[GraphData]):
             graph_index.extend([i] * num_nodes)
         self.graph_index = np.array(graph_index)
 
-        super().__init__(
-            node_features=batch_node_features,
-            edge_index=batch_edge_index,
-            edge_features=batch_edge_features,
-            node_pos_features=batch_node_pos_features,
-        )
+        # Batch user defined attributes
+        kwargs = {}
+        user_defined_attribute_names = self._get_user_defined_attributes(
+            graph_list[0])
+        for name in user_defined_attribute_names:
+            kwargs[name] = np.vstack(
+                [getattr(graph, name) for graph in graph_list])
+
+        super().__init__(node_features=batch_node_features,
+                         edge_index=batch_edge_index,
+                         edge_features=batch_edge_features,
+                         node_pos_features=batch_node_pos_features,
+                         **kwargs)
+
+    def _get_user_defined_attributes(self, graph_data: GraphData):
+        """A GraphData object can have user defined attributes but the attribute name of those
+        are unknown since it can be arbitary. This method helps to find user defined attribute's
+        name by making a list of known graph data attributes and finding other user defined
+        attributes via `vars` method."""
+        graph_data_attributes = [
```

Comment:
We should add this list to the docstring that these are the known user-defined-attributes that the class will check for

Reply:
added attributes to docstring in 9e8c7f9